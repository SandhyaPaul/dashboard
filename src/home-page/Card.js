import React from 'react';
import './card.css';

class Card extends React.PureComponent {
	render() {
		const { os, year, place, court, stage, nextCourtDate, caseType, caseSubtype } = this.props;
		const arr = [
			{ iconClass: 'fa-file-text-o', label: 'OS No', value: `${os}/${year}` },
			{ iconClass: 'fa-map-marker', label: 'Place', value: place },
			{ iconClass: 'fa-bank', label: 'Court', value: court },
			{ iconClass: 'fa-check', label: 'Stage', value: stage },
			{ iconClass: 'fa-calendar-check-o', label: 'Next Date', value: new Date(nextCourtDate).toDateString() },
			{ iconClass: 'fa-sitemap', label: 'Case Type', value: `${caseType} -> ${caseSubtype}` }
		];
		return (
			<div className="card card-raise">
				{arr.map(item => (
					<div className="card-row" key={item.label}>
						<span className="card-row-first-span">
							<i className={`fa ${item.iconClass}`} />
							<strong>{item.label}</strong>
						</span>
						<span className="card-row-second-span">{item.value}</span>
					</div>
				))}
			</div>
		);
	}
}

export { Card as TestCard };

export default Card;
