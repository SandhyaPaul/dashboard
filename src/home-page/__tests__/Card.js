import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { TestCard } from '../Card';

describe('DOM rendering', () => {
	it('renders Card', () => {
		shallow(<TestCard />);
	});

	it('DOM snapshot of Card', () => {
		const wrapper = shallow(<TestCard />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
