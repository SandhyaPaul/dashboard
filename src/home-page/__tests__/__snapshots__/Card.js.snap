// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM rendering DOM snapshot of Card 1`] = `
<div
  className="card card-raise"
>
  <div
    className="card-row"
    key="OS No"
  >
    <span
      className="card-row-first-span"
    >
      <i
        className="fa fa-file-text-o"
      />
      <strong>
        OS No
      </strong>
    </span>
    <span
      className="card-row-second-span"
    >
      undefined/undefined
    </span>
  </div>
  <div
    className="card-row"
    key="Place"
  >
    <span
      className="card-row-first-span"
    >
      <i
        className="fa fa-map-marker"
      />
      <strong>
        Place
      </strong>
    </span>
    <span
      className="card-row-second-span"
    />
  </div>
  <div
    className="card-row"
    key="Court"
  >
    <span
      className="card-row-first-span"
    >
      <i
        className="fa fa-bank"
      />
      <strong>
        Court
      </strong>
    </span>
    <span
      className="card-row-second-span"
    />
  </div>
  <div
    className="card-row"
    key="Stage"
  >
    <span
      className="card-row-first-span"
    >
      <i
        className="fa fa-check"
      />
      <strong>
        Stage
      </strong>
    </span>
    <span
      className="card-row-second-span"
    />
  </div>
  <div
    className="card-row"
    key="Next Date"
  >
    <span
      className="card-row-first-span"
    >
      <i
        className="fa fa-calendar-check-o"
      />
      <strong>
        Next Date
      </strong>
    </span>
    <span
      className="card-row-second-span"
    >
      Invalid Date
    </span>
  </div>
  <div
    className="card-row"
    key="Case Type"
  >
    <span
      className="card-row-first-span"
    >
      <i
        className="fa fa-sitemap"
      />
      <strong>
        Case Type
      </strong>
    </span>
    <span
      className="card-row-second-span"
    >
      undefined -&gt; undefined
    </span>
  </div>
</div>
`;
